packages/pages-shared/asset-server/handler.ts

```diff
@@ -446,7 +446,10 @@ export async function generateHandler<
 				};
 			}
 		);
-		const matches = headersMatcher({ request });
+
+		// If the response was an internal error from the asset handler, skip applying header rules.
+		// This avoids a case where we can cache 500s for a long time, depending on headers set by a user.
+		const matches = response.ok ? headersMatcher({ request }) : [];
```

Comment:
@WalshyDev is `response.ok` the proper check here? It's apparently false for 3XX responses.

Reply:
[response.ok](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) will only check for 2xx codes. We probably want a `response.status < 400 && response.status > 599` or similar.

Reply:
should be better: https://github.com/cloudflare/workers-sdk/compare/c2ebd1fa5cea326202e32df4728019a7ce705156..3cc829d7f12f65b7686e2f56e4b651969049691f